extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -19,9 +19,16 @@ import {listen} from '../../../src/event-helper';
 import {dev} from '../../../src/log';
 
 const TAG = 'amp-viewer-messaging';
-const sentinel_ = '__AMPHTML__';
-const requestSentinel_ = sentinel_ + 'REQUEST';
-const responseSentinel_ = sentinel_ + 'RESPONSE';
+const APP = '__AMPHTML__';
```

Comment:
I'll let @dvoytenko do the real review, but one request:

The current viewer integration writes to a magic global variable `self.viewerState` with its major progress and error messages. This allows the error reporting to transport that context info. We should retain that functionality.